mlcomp/parallelm/pipeline/component_info.py

```diff
@@ -0,0 +1,319 @@
+                ret[json_fields.COMPONENT_DESC_INPUT_INFO_FIELD] = obj.inputs
+            if obj.outputs:
+                ret[json_fields.COMPONENT_DESC_OUTPUT_INFO_FIELD] = obj.outputs
+            if obj.arguments:
+                ret[json_fields.COMPONENT_DESC_ARGUMENTS] = obj.arguments
+            return ret
+        elif isinstance(obj, ComponentConnectionInfo):
+            ret = OrderedDict()
+            if obj.description:
+                ret[json_fields.CONNECTION_DESC_DESCRIPTION_FIELD] = obj.description
+            if obj.label:
+                ret[json_fields.CONNECTION_DESC_LABEL_FIELD] = obj.label
+            if obj.default_component:
+                ret[json_fields.CONNECTION_DESC_DEFAULT_COMPONENT_FIELD] = obj.default_component
+            if obj.type:
+                ret[json_fields.CONNECTION_DESC_TYPE_FIELD] = obj.type
+            if obj.group:
+                ret[json_fields.CONNECTION_DESC_GROUP_FIELD] = obj.group
+            return ret
+        elif isinstance(obj, ComponentArgumentInfo):
+            ret = OrderedDict()
+            if obj.key:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_KEY] = obj.key
+            if obj.type:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_TYPE] = obj.type
+            if obj.label:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_LABEL] = obj.label
+            if obj.description:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_DESCRIPTION] = obj.description
+            if obj.env_var:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_ENV_VAR] = obj.env_var
+            if obj.optional:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_OPTIONAL] = obj.optional
+            if obj.tag:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_TAG] = obj.tag
+            if obj.default_value:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_DEFAULT_VAL] = obj.default_value
+            return ret
+        else:
+            # Let the base class default method raise the TypeError
+            return json.JSONEncoder.default(self, obj)
+
+
+class ComponentInfo(object):
+    def __init__(self, comp_desc_json=None):
+        self._name = None
+        self._engine_type = None
+        self._language = None
+        self._group = None
+        self._label = None
+        self._description = None
+        self._version = None
+        self._user_standalone = None
+        self._program = None
+        self._component_class = None
+        self._model_behavior = None
+        self._use_mlops = None
+        self._deps = None
+        self._include_glob_patterns = None
+        self._exclude_glob_patterns = None
+        self._inputs = None
+        self._outputs = None
+        self._arguments = None
+
+        self.load_from_json(comp_desc_json)
+
+    def load_from_json(self, comp_desc_json):
+        if comp_desc_json:
+            self.version = comp_desc_json.get(json_fields.COMPONENT_DESC_VERSION_FIELD)
```

Comment:
Is it supposed to be `self._version`? (Same for all other fields)
Why not using the `comp_desc_json` directly?

Reply:
What do you mean? Assign `comp_desc_json` to `__dict__`?